proto/clustermanagerdata.proto

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2018 The Vitess Authors
+ Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+     http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreedto in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This file contains vitess cluster manager related data structures.
+
+syntax = "proto3";
+option go_package = "vitess.io/vitess/go/vt/proto/clustermanagerdata";
+
+option java_package="io.vitess.proto";
+
+import "topodata.proto";
+
+package clustermanagerdata;
+
+// SetShardConfigRequest is the payload for SetShardConfigRequest
+message SetShardConfigRequest {
+  string cell = 1;
+  string keyspace = 2;
+  string shard = 3;
+  topodata.TabletType tablet_type = 4;
+  // number of nodes desired for (keyspace/shard/cell/tablet_type)
+  int32 nodes = 5;
+}
+
+// DeleteShardConfigRequest is the payload for DeleteShardConfig
+message DeleteShardConfigRequest{
+  string cell = 1;
+  string keyspace = 2;
+  string shard = 3;
+  string tablet_type = 4;
+}
+
+// DeleteShardConfigResponse is the returned value for DeleteShardConfigRequest.
+message DeleteShardConfigResponse{}
+
+// GetClustrConfigRequest is the paylaod for GetClusterConfig.
+message GetClusterConfigRequest{
+  // cell is optional
+  string cell = 1;
+  // keyspace is optional, if cell is provided
+  string keyspace = 2;
+  // shard is optional, if cell/keyspace is provided
+  string shard = 3;
+  // tablet_type is optional, if cell/keyspace/shard are provided
+  string tablet_type = 4;
+}
+
+// ClusterConfig defines the configuration for a vitess cluster.
+message ClusterConfig{
```

Comment:
Organizationally, it makes more sense to me to nest the below configs under ClusterConfig. 

https://developers.google.com/protocol-buffers/docs/proto#nested

What do you think?